test/integration/good/code-gen/cpp.expected

```diff
@@ -38,28 +47,27 @@ class eight_schools_ncp_model : public prob_grad {
     (void) function__;  // suppress unused var warning
     
     
-    stan::model::assign(J, stan::model::nil_index_list(), context__.vals_i("J")
-                                                          [(1 - 1)], "assigning variable J");
+    assign(J, nil_index_list(), context__.vals_i("J")[(1 - 1)], "assigning variable J");
     check_greater_or_equal(function__, "J", J, 0);
     y = std::vector<double>(J, 0);
     
     pos__ = 1;
     for (size_t sym1__ = 1; sym1__ <= J; ++sym1__) {
-      stan::model::assign(y, stan::model::cons_list(stan::model::index_uni(sym1__), stan::model::nil_index_list()), 
-                          context__.vals_r("y")[(pos__ - 1)], "assigning variable y[(sym1__ - 1)]");
-      stan::model::assign(pos__, stan::model::nil_index_list(), (pos__ + 1), "assigning variable pos__");
+      assign(y, cons_list(index_uni(sym1__), nil_index_list()), context__.vals_r("y")
+                                                                [(pos__ - 1)], "assigning variable y[(sym1__ - 1)]");
+      assign(pos__, nil_index_list(), (stan::model::deep_copy(pos__) + 1), "assigning variable pos__");
     }
     sigma = std::vector<double>(J, 0);
     
     pos__ = 1;
     for (size_t sym1__ = 1; sym1__ <= J; ++sym1__) {
-      stan::model::assign(sigma, stan::model::cons_list(stan::model::index_uni(sym1__), stan::model::nil_index_list()), 
-                          context__.vals_r("sigma")[(pos__ - 1)], "assigning variable sigma[(sym1__ - 1)]");
-      stan::model::assign(pos__, stan::model::nil_index_list(), (pos__ + 1), "assigning variable pos__");
+      assign(sigma, cons_list(index_uni(sym1__), nil_index_list()), context__.vals_r("sigma")
+                                                                    [(pos__ - 1)], "assigning variable sigma[(sym1__ - 1)]");
+      assign(pos__, nil_index_list(), (stan::model::deep_copy(pos__) + 1), "assigning variable pos__");
```

Comment:
I remember that the old compiler would warn on making deep copies. Is that a runtime warning that the C++ currently also gives or would it be a compile time warning whenever `deep_copy` code is generated?

Reply:
And this is a particularly ugly place for a deepcopy, given that `pos__` is a compiler-generated integer lol. I can fix this particular one by checking the type to see that it's not an int. 

It's not a runtime warning - I think it probably needs to become stanc3-compile-time warning. Let's talk about it at the meeting and see how we should incorporate that.